pfcpiface/up4.go

```diff
@@ -369,10 +381,38 @@ func (up4 *UP4) setUpfInfo(u *upf, conf *Conf) {
 		up4.counters[i].allocated = make(map[uint64]uint64)
 	}
 
-	err := up4.tryConnect()
+	go up4.keepTryingToConnect()
+}
+
+func (up4 *UP4) tryConnect() error {
+	if up4.isConnected(nil) {
+		return nil
+	}
+
+	err := up4.setupChannel()
 	if err != nil {
-		log.Errorf("failed to connect to UP4: %v", err)
-		return
+		return err
+	}
+
+	err = up4.initialize()
+	if err != nil {
+		log.Fatalf("Failed to initialize UP4: %v", err)
+	}
+
+	up4.setConnectedStatus(true)
+
+	return nil
+}
+
+func (up4 *UP4) keepTryingToConnect() {
+	for {
+		err := up4.tryConnect()
```

Comment:
Can this create a race with `tryConnect` in `sendMsgToUPF`?

Reply:
`tryConnectMu` prevents from race condition now. I believe it's useful to reconnect periodically. With the "lazy" setup we will face the same issue as we did so far -> if there is no SMF and ONOS is restarted at some point, pfcp-agent will not reconnect.